development-docs/systemtests/io.strimzi.systemtest.cruisecontrol.CruiseControlST.md

```diff
@@ -0,0 +1,218 @@
+| 5. | Ensure that Cruise Control pod is rolled due to configuration change | Cruise Control pod is rolled and new configuration is applied |
+| 6. | Verify the updated Cruise Control configuration | Updated replica movement strategies are verified in Cruise Control configuration |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlTopicExclusion
+
+**Description:** Verify that Kafka Cruise Control excludes specified topics and includes others.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Set topic names | Topic names are set |
+| 2. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 3. | Deploy Kafka with Cruise Control enabled | Kafka cluster with Cruise Control is deployed |
+| 4. | Create topics to be excluded and included | Topics 'excluded-topic-1', 'excluded-topic-2', and 'included-topic' are created |
+| 5. | Create KafkaRebalance resource excluding specific topics | KafkaRebalance resource is created with 'excluded-.*' topics pattern |
+| 6. | Wait for KafkaRebalance to reach ProposalReady state | KafkaRebalance reaches the ProposalReady state |
+| 7. | Check optimization result for excluded and included topics | Excluded topics are in the optimization result and included topic is not |
+| 8. | Approve the KafkaRebalance proposal | KafkaRebalance proposal is approved |
+| 9. | Wait for KafkaRebalance to reach Ready state | KafkaRebalance reaches the Ready state |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlWithApiSecurityDisabled
+
+**Description:** Test the Cruise Control functionality when API security is disabled.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 2. | Create a Kafka cluster with Cruise Control having API security disabled | Kafka cluster with Cruise Control and no API security is deployed |
+| 3. | Create a Kafka Rebalance resource | Kafka Rebalance resource is successfully created |
+| 4. | Wait for the Kafka Rebalance custom resource state to become ProposalReady | Kafka Rebalance custom resource state is ProposalReady |
+
+**Labels:**
+
+* [cruise-control](labels/cruise-control.md)
+
+
+## testCruiseControlWithRebalanceResourceAndRefreshAnnotation
+
+**Description:** Using Kafka cluster within a single namespace to test Cruise Control with rebalance resource and refresh annotation.
+
+**Steps:**
+
+| Step | Action | Result |
+| - | - | - |
+| 1. | Create broker and controller KafkaNodePools. | Both KafkaNodePools are successfully created |
+| 2. | Create Kafka cluster | Kafka cluster with ephemeral storage is created and available |
+| 3. | Deploy Kafka Rebalance resource | Kafka Rebalance resource is deployed and in NotReady state |
+| 4. | Enable Cruise Control with tuned spec | Cruise Control is enabled and configured |
```

Comment:
Not sure what we mean by `tuned spec` here? Maybe we should be more specific?

Reply:
not sure we want to mention all these options 
```
.editOrNewCruiseControl()
                    .addToConfig("max.active.user.tasks", 10)
                    .addToConfig("metric.sampling.interval.ms", 5_000)
                    .addToConfig("cruise.control.metrics.reporter.metrics.reporting.interval.ms", 5_000)
                    .addToConfig("metadata.max.age.ms", 4_000)
                    .addToConfig("sample.store.topic.replication.factor", 1)
                    .addToConfig("partition.sample.store.topic.partition.count", 1)
                    .addToConfig("broker.sample.store.topic.partition.count", 1)
                    .addToConfig("skip.sample.store.topic.rack.awareness.check", true)
                    .addToConfig("partition.metrics.window.ms", 10_000)
                    .addToConfig("broker.metrics.window.ms", 10_000)
                    .addToConfig("monitor.state.update.interval.ms", 10_000)
```
:D

Reply:
optimized configuration?

Reply:
Sure I will update it.